planner/core/task.go

```diff
@@ -870,6 +881,14 @@ func (p *basePhysicalAgg) newPartialAggregate(copToFlash bool) (partial, final P
 	sc := p.ctx.GetSessionVars().StmtCtx
 	client := p.ctx.GetClient()
 	for _, aggFunc := range p.AggFuncs {
+		// Pre-allocate a slice to reduce allocation, 8 doesn't have special meaning.
+		result := make([]*expression.Column, 0, 8)
+		cols := expression.ExtractColumnsFromExpressions(result, aggFunc.Args, nil)
```

Comment:
Could you add a method directly chech whether there's gen cols, without any slice allocating?

Reply:
done!